src/components/SignIn/SignIn.js

```diff
@@ -1,29 +1,29 @@
 import PopupWithForm from "../PopupWithForm/PopupWithForm";
-import { useLoggedIn } from "../../context/LoggedInContext";
-import { useNavigate } from "react-router-dom";
-import { useState } from "react";
+import { useFormWithValidation } from "../../utils/useForm";
 
 function SignIn({ isPopupOpen, onClose, onRedirect, handlePopupSubmit }) {
-  const { setIsLoggedIn } = useLoggedIn();
-  const navigate = useNavigate();
-
-  const [email, setEmail] = useState("");
-  const [password, setPassword] = useState("");
-
-  const validate = () => {
-    return email.length > 0 && password.length > 0;
-  };
+  const {
+    values,
+    handleChange,
+    errors,
+    isValid,
+    resetForm,
+    serverError,
+    setServerError,
+  } = useFormWithValidation();
 
```

Comment:
* The app crashes when I go to `Saved Articles` if I'm not authorized. See here https://disk.yandex.ru/i/Ygz1tgoScQsvSA but the criteria say: **When trying to access the `/saved-news` route, unauthorized users are redirected to `/signin` with an open authorization popup window.**